credentials being populated incorrectly

```
$ cat ~/.local/share/juju/credentials.yaml 
credentials:
  aws:
    null:
      access-key: xxxxxxxx
      auth-type: access-key
      secret-key: xxxxxxxx

```
And here's the conjure-up log.
```
$ cat ~/.cache/conjure-up/conjure-up.log 
2017-06-02 10:17:43,727 [DEBUG] conjure-up/_unspecified_spell - app.py:216 - LXD version: 2.13, Juju version: 2.2-rc1-xenial-amd64, conjure-up version: 2.2-rc1
2017-06-02 10:17:43,737 [DEBUG] conjure-up/_unspecified_spell - app.py:272 - found spell {'key': 'canonical-kubernetes', 'name': 'The Canonical Distribution of Kubernetes', 'description': 'Kubernetes is an open-source platform for deploying, scaling, and operations of application containers across a cluster of hosts. Kubernetes is portable in that it works with public, private, and hybrid clouds. Extensible through a pluggable infrastructure. Self healing in that it will automatically restart and place containers on healthy nodes if a node ever goes away.\n'}
2017-06-02 10:17:43,738 [DEBUG] conjure-up/canonical-kubernetes - download.py:60 - Path is local filesystem, copying /snap/conjure-up/355/spells/canonical-kubernetes to /home/ericgf/.cache/conjure-up/canonical-kubernetes
2017-06-02 10:17:43,744 [INFO] conjure-up/canonical-kubernetes - events.py:156 - Watching for shutdown
2017-06-02 10:17:43,771 [DEBUG] conjure-up/canonical-kubernetes - events.py:50 - Awaiting Shutdown at conjureup/events.py:158
2017-06-02 10:17:43,772 [DEBUG] conjure-up/canonical-kubernetes - utils.py:348 - Pulling bundle for canonical-kubernetes from channel: stable
2017-06-02 10:18:06,163 [INFO] conjure-up/canonical-kubernetes - common.py:71 - Bootstrapping Juju controller.
2017-06-02 10:18:06,164 [DEBUG] conjure-up/canonical-kubernetes - juju.py:139 - Bootstrapping to set region: {}
2017-06-02 10:18:06,166 [DEBUG] conjure-up/canonical-kubernetes - juju.py:170 - bootstrap cmd: ['juju', 'bootstrap', 'aws/us-east-2', 'conjure-up-aws-b84', '--default-model', 'conjure-canonical-kubern-17e', '--config', 'image-stream=daily', '--config', 'enable-os-upgrade=false', '--bootstrap-series', 'xenial']
2017-06-02 10:18:06,181 [DEBUG] conjure-up/canonical-kubernetes - events.py:50 - Awaiting ModelConnected at conjureup/controllers/deploy/common.py:22
2017-06-02 10:18:06,206 [DEBUG] conjure-up/canonical-kubernetes - juju.py:179 - waiting for proc
2017-06-02 10:18:06,207 [DEBUG] conjure-up/canonical-kubernetes - juju.py:181 - proc done
2017-06-02 10:18:06,208 [ERROR] conjure-up/canonical-kubernetes - common.py:35 - Error bootstrapping controller: ['ERROR loading credentials: credentials.aws: expected string, got nothing']
2017-06-02 10:18:06,208 [DEBUG] conjure-up/canonical-kubernetes - events.py:50 - Setting Error at conjureup/events.py:133
2017-06-02 10:18:06,209 [ERROR] conjure-up/canonical-kubernetes - events.py:141 - Unhandled exception
Traceback (most recent call last):
  File "/snap/conjure-up/355/lib/python3.6/site-packages/conjureup/events.py", line 139, in handle_exception
    raise exc
  File "/snap/conjure-up/355/lib/python3.6/site-packages/conjureup/controllers/bootstrap/common.py", line 37, in do_bootstrap
    app.current_cloud_type))
Exception: Unable to bootstrap (cloud type: ec2)
2017-06-02 10:18:06,210 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:32 - Showing dialog for exception: Unable to bootstrap (cloud type: ec2)
2017-06-02 10:18:09,397 [DEBUG] conjure-up/canonical-kubernetes - events.py:50 - Setting Shutdown at conjureup/events.py:129
2017-06-02 10:18:09,402 [DEBUG] conjure-up/canonical-kubernetes - events.py:50 - Received Shutdown at conjureup/events.py:158
2017-06-02 10:18:09,402 [INFO] conjure-up/canonical-kubernetes - events.py:162 - Shutting down
```

--- a/conjureup/controllers/credentials/gui.py
+++ b/conjureup/controllers/credentials/gui.py
@@ -67,6 +67,8 @@ class CredentialsController(common.BaseCredentialsController):
 
     def save_credential(self, credential):
         cred_path = path.join(utils.juju_path(), 'credentials.yaml')
+        cred_name = "conjure-{}-{}".format(app.current_cloud, utils.gen_hash())
+
         try:
             existing_creds = yaml.safe_load(open(cred_path))
         except:
@@ -74,12 +76,12 @@ class CredentialsController(common.BaseCredentialsController):
 
         if app.current_cloud in existing_creds['credentials'].keys():
             c = existing_creds['credentials'][app.current_cloud]
-            c[app.current_controller] = self._format_creds(credential)
+            c[cred_name] = self._format_creds(credential)
         else:
             # Handle the case where path exists but an entry for the cloud
             # has yet to be added.
             existing_creds['credentials'][app.current_cloud] = {
-                app.current_controller: self._format_creds(credential)
+                cred_name: self._format_creds(credential)
             }
 
         with open(cred_path, 'w') as cred_f:
@@ -94,9 +96,9 @@ class CredentialsController(common.BaseCredentialsController):
                 juju.add_cloud(app.current_cloud,
                                credential.cloud_config())
 
-        # This should return the app.current_controller so juju bootstrap knows
+        # This should return the credential name so juju bootstrap knows
         # which credential to bootstrap with
-        self.finish(app.current_controller)
+        self.finish(cred_name)
 
 
 _controller_class = CredentialsController

